enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -1081,9 +1087,11 @@ void calculateUnusedValuesInFunction(
     for (auto &BB : func)
       for (auto &I : BB) {
         bool ivn = DifferentialUseAnalysis::is_value_needed_in_reverse<
-            ValueType::Primal>(gutils, &I, mode, PrimalSeen, oldUnreachable);
+            ValueType::Primal>(gutils, &I, mode, PrimalSeen,
+                               gutils->notForAnalysis);
```

Comment:
Could we get the `mode` and `oldUnreachable` from `gutils` instead of passing each to the constructor?